Documentation/sources/content-card-ui/public-classes/aepui.md

```diff
@@ -0,0 +1,96 @@
+# Sealed Interface - AepUI
+
+The `AepUI` interface represents a UI component that can be rendered using the AEP compose UI library. In the initial release of the AEP compose UI library, the `SmallImageUI` implements an `AepUI`  to display a Small Image Template. The `AepUI` associates with a specific template and state to ensure consistency when rendering and managing UI components.
+## Methods
+
+### getTemplate 
+
+Retrieves the template associated with this UI component.
+
+#### Returns
+
+A template of type `T` associated with the UI componenet.
+
+#### Syntax
+
+``` kotlin
+fun getTemplate(): T
+```
+
+### getState 
+
+Retrieves the current state of the UI component.
+
+#### Returns
+
+A state of type `S` associated with the UI componenet.
```

Comment:
nit: typo componenet -> component
Can we also specify that S would be an implementation of `AepCardUIState` interface

Reply:
Thinking about this a bit more, @prudrabhat do you think we should call this interface `AepUIState` instead of `AepCardUIState` to be flexible to extend for non card type UIs?

Reply:
We can do it in a separate PR after discussing offline 👍